src/python/dxpy/utils/executable_unbuilder.py

```diff
@@ -115,85 +116,113 @@ def make_cluster_bootstrap_script_file(region, entry_point, code, suffix):
+    except DXError:
+        print("Failed to get permitted regions where {} can perform billable activities.\n".format(bill_to), file=sys.stderr)
+        sys.exit(1)    
+    
+    enabled_regions = set(info["runSpec"]["bundledDependsByRegion"].keys())
+    if not enabled_regions.issubset(permitted_regions):
+        print("Region(s) {} are not among the permitted regions of {}. Resources from these regions will not be available.".format(
+            ", ".join(enabled_regions.difference(permitted_regions)), bill_to))
+    # Update enabled regions
+    enabled_regions.intersection_update(permitted_regions)
+    # Check if at least one region is enabled when not omitting resources
     if not omit_resources:
-        for dep in info["runSpec"]["bundledDepends"]:
-            if dep in deps_to_remove:
-                continue
-            handler = get_handler(dep["id"])
-            if isinstance(handler, dxpy.DXFile):
+        if not enabled_regions:
+            raise DXError(
+                "Cannot download resources of the requested executable {} since it is not available in any of the billable regions. "
+                "You can use the --omit-resources flag to skip downloading the resources. ".format(info["name"]))
+        
+        print("Dependencies could be retrieved from region: {}. ".format(", ".join(enabled_regions)))
+        # Pick a source region. The current selected region is preferred
+        try:
+            current_region = dxpy.api.project_describe(dxpy.WORKSPACE_ID, input_params={"fields": {"region": True}})["region"]
+        except:
+            current_region = None
+
+        if current_region in enabled_regions:
+            source_region  = current_region
+            print("Trying to download resources from the current region {}...".format(source_region))
+        else:
+            source_region = enabled_regions.pop()
+            print("Trying to download resources from one of the enabled region {}...".format(source_region))
+
+        # When an app(let) is built the following dependencies are added as bundledDepends:
+        # 1. bundledDepends explicitly specified in the dxapp.json
+        # 2. resources (contents of resources directory added as bundledDepends)
+        # 3. assetDepends in the dxapp.json (with their record IDs translated into file IDs)
+        #
+        # To get the resources, we will tranverse the bundleDepends list in the source region and do the following:
+        # - If an file ID refers to an AssetBundle tarball (a file with the "AssetBundle" property), 
+        #   skip downloading the file, and keep this file ID as a bundledDepends in the final dxapp.json
+        # - Otherwise, download the file and remove this ID from the bundledDepends list in the final dxapp.json
+
+        def untar_strip_leading_slash(tarfname, path):
+            t = tarfile.open(tarfname)
+            for m in t.getmembers():
+                if m.name.startswith("/"):
```

Comment:
There is `lstrip` method, which removes characters from the beginning of the string if there are such characters, otherwise returns the original string. 

(`m.name.lstrip("/")` would remove all leading slashes from the name - `//string` would become `string`)

there is also `removeprefix` method, which removes an exact prefix in case it is there, but it was added only in python 3.9.